### PR TITLE
chore: replace pylint and black with ruff

### PR DESCRIPTION
**Pull request by Venefilyn**
Ruff is a drop-in replacement for black and also handles linting
for us in a lot nicer way than pylint. More specifically it is a lot
faster than the alternatives and allows us to enforce PyFormat
for strings rather than percent-format or f-strings.

Jira Issues:

-

Checklist

- [ ] PR has been tested manually in a VM (either author or reviewer)
- [ ] Jira issue has been made public if possible
- [ ] `[RHELC-]` or `[HMS-]` is part of the PR title 
- [ ] GitHub label has been added to help with Release notes 
- [ ] PR title explains the change from the user's point of view
- [ ] Code and tests are documented properly
- [ ] The commits are squashed to as few commits as possible (without losing data) 
- [ ] When merged: Jira issue has been updated to `Release Pending` if relevant

### CODE REVIEW

File: tests/integration/tier0/non-destructive/config-file/test_config_file.py [hunk 1]
```diff
@@ -44,15 +44,7 @@ def c2r_config_setup(request, backup_directory):
...
-            # We disable the Black pre-commit hook because it changes formatting
-            # of the following configuration files, which makes them hard to read.
-            # https://black.readthedocs.io/en/stable/usage_and_configuration/the_basics.html#ignoring-sections
-            # fmt: off
-            (
-                "[subscription_manager]\n"
-                "password = config_password\n"
-            ),
-            # fmt: on
+            ("[subscription_manager]\n" "password = config_password\n"),
```
Comment (kokesak) [inline tests/integration/tier0/non-destructive/config-file/test_config_file.py:47]: Is there a possibility to disable the ruff for a specific block of code? I would like to keep this to improve the readability. The only thing I found is this https://www.reddit.com/r/neovim/comments/1aeinr5/how_to_make_ruff_ignore_one_code_block/, but it suggests what is already in the code -> to use the `fmt: off`

Comment (Venefilyn) [reply tests/integration/tier0/non-destructive/config-file/test_config_file.py:47]: Ah you're right, should probably move this back

https://docs.astral.sh/ruff/formatter/#format-suppression

File: pyproject.toml [hunk 2]
```diff
@@ -1,22 +1,30 @@
...
+[tool.ruff]
+builtins = ["convert2rhel", "six"]
+
+
+line-length = 120
+fix = true
+
+# We technically only support py27 py36 due to el7 and el8 but ruff only
+# supports py37 as minimum. This is not a huge issue as any rules we do not
+# want we can disablew
```
Comment (hosekadam) [inline pyproject.toml:17]:
```suggestion
# want we can disable
```

Comment (Venefilyn) [reply pyproject.toml]: Thanks, fixed

File: tests/integration/tier0/non-destructive/config-file/test_config_file.py (hunk 1 above)
Comment (Venefilyn) [reply tests/integration/tier0/non-destructive/config-file/test_config_file.py]: Fixed it  @kokesak

Comment (kokesak) [reply tests/integration/tier0/non-destructive/config-file/test_config_file.py]: Thanks!

**Review by kokesak — APPROVED** [review]

**Review by hosekadam — APPROVED** [review]

**Comment by r0x0d** [comment]
I've added the sanity test label just to be sure that nothing is broken. It will not, but just in case. Doesn't hurt to run the sanity tests suite

File: tests/integration/common/checks-after-conversion/test_verify_strings_in_log.py [hunk 3]
```diff
@@ -1,7 +1,5 @@
 import re
 
-from test_helpers.common_functions import log_file_data
-
```
Comment (kokesak) [inline tests/integration/common/checks-after-conversion/test_verify_strings_in_log.py:4]: Do not remove this line. It is needed in the test

Comment (Venefilyn) [reply tests/integration/common/checks-after-conversion/test_verify_strings_in_log.py:4]: Couldn't we just move it to conftest to have it be available without import? It doesn't feel good to have to import a fixture like this as it is a bit confusing. I'd rather have pytest take care of it in something like conftest
```diff
diff --git a/tests/integration/conftest.py b/tests/integration/conftest.py
index bfb0596b..f299bc1c 100644
--- a/tests/integration/conftest.py
+++ b/tests/integration/conftest.py
@@ -24,12 +24,12 @@ except ImportError:
 #
 pytest.register_assert_rewrite("test_helpers")
 
-from test_helpers.common_functions import SystemInformationRelease, get_full_kernel_title  # noqa: E402
-from test_helpers.satellite import Satellite  # noqa: E402
-from test_helpers.shell import live_shell  # noqa: E402
-from test_helpers.subscription_manager import SubscriptionManager  # noqa: E402
-from test_helpers.vars import SYSTEM_RELEASE_ENV, TEST_VARS  # noqa: E402
-from test_helpers.workarounds import workaround_grub_setup  # noqa: E402
+from test_helpers.common_functions import SystemInformationRelease, get_full_kernel_title
+from test_helpers.satellite import Satellite
+from test_helpers.shell import live_shell
+from test_helpers.subscription_manager import SubscriptionManager
+from test_helpers.vars import SYSTEM_RELEASE_ENV, TEST_VARS
+from test_helpers.workarounds import workaround_grub_setup
 
 
 logging.basicConfig(level=os.environ.get("DEBUG", "INFO"), stream=sys.stderr)
@@ -538,3 +538,20 @@ def backup_directory(shell, request):
     yield backup_path
 
     shell(f"rm -rf {backup_path}")
+
+
+@pytest.fixture()
+def log_file_data():
+    """
+    Helper fixture.
+    Reads and returns data from the convert2rhel.log file.
+    Mainly used for after conversion checks, where we match required strings to the log output.
+    """
+    convert2rhel_log_file = "/var/log/convert2rhel/convert2rhel.log"
+
+    assert os.path.exists(convert2rhel_log_file), "The c2r log file does not exits."
+
+    with open(convert2rhel_log_file, "r") as logfile:
+        log_data = logfile.read()
+
+        return log_data
diff --git a/tests/integration/test_helpers/common_functions.py b/tests/integration/test_helpers/common_functions.py
index 069626b1..c2cd9e07 100644
--- a/tests/integration/test_helpers/common_functions.py
+++ b/tests/integration/test_helpers/common_functions.py
@@ -108,20 +108,3 @@ def get_custom_repos_names():
                 f"rhel-{system_version.major}-for-x86_64-appstream-rpms",
             ]
     return repos
-
-
-@pytest.fixture()
-def log_file_data():
-    """
-    Helper fixture.
-    Reads and returns data from the convert2rhel.log file.
-    Mainly used for after conversion checks, where we match required strings to the log output.
-    """
-    convert2rhel_log_file = "/var/log/convert2rhel/convert2rhel.log"
-
-    assert os.path.exists(convert2rhel_log_file), "The c2r log file does not exits."
-
-    with open(convert2rhel_log_file, "r") as logfile:
-        log_data = logfile.read()
-
-        return log_data
```

**Comment by Venefilyn** [comment]
@kokesak All integration tests are passing now with the change I made 

```py
# We use pytest_plugins to allow us to use fixtures we define in other files without the need to explicitly import them
# inside each test file.
# LINK - https://docs.pytest.org/en/7.0.x/reference/reference.html#globalvar-pytest_plugins
pytest_plugins = (
    "test_helpers.common_functions",
    "test_helpers.workarounds",
)
```

As for unit tests failing, it is unclear to me at the moment what the underlying issue is with the encoding tests in el7

But for the failing test relating to restorable file it is unrelated to any change I've made. It is just a unit test previously not being executed due to wrong name. Unsure if the error is the test or implementation atm

@r0x0d you are a code owner as well atm, wanna give a quick review of the devcontainer stuff that's changed?

**Review by kokesak — APPROVED** [review]
QE ack

**Review by r0x0d — APPROVED** [review]

**Pull request merged**